src/Compilers/Core/Portable/Text/ChangedText.cs

```diff
@@ -240,149 +238,326 @@ private static ImmutableArray<TextChangeRange> Merge(IReadOnlyList<ImmutableArra
             return merged;
         }
 
-        private static ImmutableArray<TextChangeRange> Merge(ImmutableArray<TextChangeRange> oldChanges, ImmutableArray<TextChangeRange> newChanges)
+        /// <summary>
+        /// Represents a newChange being processed by <see cref="Merge(ImmutableArray&lt;TextChangeRange&gt;, ImmutableArray&lt;TextChangeRange&gt;)"/>.
+        /// Such a newChange must be adjusted before being added to the result list.
+        /// </summary>
+        private readonly struct UnadjustedNewChange
```

Comment:
might want to explicit mention that it's like  TextChangeRange, but has to represent more concepts (like negative moes).

Reply:
Done, thanks!